edalize/nextpnr.py

```diff
@@ -95,37 +95,53 @@ def configure_main(self):
         file_table = []
 
         output_format = self.tool_options.get('output_format', 'fasm')
-        arch = self.tool_options.get('arch', 'xilinx')
+        arch = self.tool_options.get('arch')
+
+        assert arch in ['xilinx', 'fpga_interchange'], 'Missing or invalid "arch" parameter in "tool_options"'
 
         synth_design = None
         chipdb = None
+        device = None
         xdc = None
 
         for f in src_files:
             if f.file_type in ['synthJson']:
                 synth_design = f.name
             elif f.file_type in ['bba']:
                 chipdb = f.name
+            elif f.file_type in ['device']:
+                device = f.name
             elif f.file_type in ['xdc']:
                 xdc = f.name
             else:
                 continue
 
         assert chipdb and xdc, "Missing required files."
+        assert device is not None or arch is not 'fpga_interchange', 'Missing required ".device" file for "fpga_interchange" arch'
 
         additional_options = self.tool_options.get('nextpnr_impl_options', '')
+        package = self.tool_options.get('package', None)
+
+        assert package is not None, 'Missing required "package" parameter'
```

Comment:
Is this required for non-fpga_interchange arch as well?

Reply:
nope, we should only check this param for `fpga_interchange` arch